### PR TITLE
recipes-graphics: Add package in mesa-dev to install CI files

### PR DESCRIPTION
**Pull request by alimon**
Mesa has .gitlab-ci folder that contains scripts and fail/skips useful
in order to run tests.

Signed-off-by: Aníbal Limón <anibal.limon@linaro.org>

### CODE REVIEW

File: recipes-graphics/mesa/mesa_git.bb [hunk 1]
```diff
@@ -20,3 +20,16 @@ PV = "20.3-dev+git${SRCPV}"
...
+PACKAGES += "mesa-ci"
+FILES_${PN}-ci = "${datadir}/mesa-ci/"
+do_install_append () {
+    install -d ${D}/${datadir}/mesa-ci/
+
+    install -m 0644 ${S}/.gitlab-ci/deqp-default-skips.txt ${D}/${datadir}/mesa-ci/
+    for f in ${S}/.gitlab-ci/deqp-freedreno-*; do
+        install -m 0644 $f ${D}/${datadir}/mesa-ci/
+    done
+    install -m 0755 ${S}/.gitlab-ci/deqp-runner.sh ${D}/${datadir}/mesa-ci/
```
Comment (ndechesne) [inline recipes-graphics/mesa/mesa_git.bb:34]: maybe bindir here?

File: recipes-graphics/mesa/mesa_git.bb [hunk 2]
```diff
@@ -20,3 +20,16 @@ PV = "20.3-dev+git${SRCPV}"
 # Do not select this version by default
 DEFAULT_PREFERENCE = "-1"
 DEFAULT_PREFERENCE_sm8250 = "1"
+
+# Add package to install require files to run tests for mesa
+PACKAGES += "mesa-ci"
+FILES_${PN}-ci = "${datadir}/mesa-ci/"
```
Comment (ndechesne) [inline recipes-graphics/mesa/mesa_git.bb:26]: maybe mesa instead of mesa-ci?

Comment (alimon) [reply recipes-graphics/mesa/mesa_git.bb:26]: So in that case /usr/share/mesa/ci ?

File: recipes-graphics/mesa/mesa_git.bb (hunk 1 above)
Comment (alimon) [reply recipes-graphics/mesa/mesa_git.bb:34]: Ack.

File: recipes-graphics/mesa/mesa_git.bb [hunk 3]
```diff
@@ -20,3 +20,19 @@ PV = "20.3-dev+git${SRCPV}"
 # Do not select this version by default
 DEFAULT_PREFERENCE = "-1"
 DEFAULT_PREFERENCE_sm8250 = "1"
+
+# Add package to install require files to run tests for mesa
+PACKAGES += "mesa-ci"
+FILES_${PN}-ci = "${datadir}/mesa/ci/"
+do_install_append () {
+    install -d ${D}/${datadir}/mesa
```
Comment (ndechesne) [inline recipes-graphics/mesa/mesa_git.bb:28]: sorry, i missed that earlier, but this line is not needed. install -d will create all directories at once.

File: recipes-graphics/mesa/mesa_git.bb [hunk 4]
```diff
@@ -20,3 +20,19 @@ PV = "20.3-dev+git${SRCPV}"
...
+
+    install -d ${D}/${datadir}/mesa/ci
```
Comment (ndechesne) [inline recipes-graphics/mesa/mesa_git.bb:30]: you could install in ${datadir}/mesa directly, i don't think we need the 'ci' folder.

File: recipes-graphics/mesa/mesa_git.bb [hunk 5]
```diff
@@ -20,3 +20,19 @@ PV = "20.3-dev+git${SRCPV}"
...
+    install -m 0644 ${S}/.gitlab-ci/deqp-default-skips.txt ${D}/${datadir}/mesa/ci/
+    for f in ${S}/.gitlab-ci/deqp-freedreno-*; do
+        install -m 0644 $f ${D}/${datadir}/mesa/ci/
+    done
+
+    install -d ${D}/${datadir}/mesa/ci/bin
+    install -m 0755 ${S}/.gitlab-ci/deqp-runner.sh ${D}/${datadir}/mesa/ci/bin
```
Comment (ndechesne) [inline recipes-graphics/mesa/mesa_git.bb:37]: i meant to install in ${D}/${bindir}, not in datadir. binary should be installed in bindir.

Comment (alimon) [reply recipes-graphics/mesa/mesa_git.bb]: Ok.

File: recipes-graphics/mesa/mesa_git.bb (hunk 4 above)
Comment (alimon) [reply recipes-graphics/mesa/mesa_git.bb]: Ack.

File: recipes-graphics/mesa/mesa_git.bb (hunk 3 above)
Comment (alimon) [reply recipes-graphics/mesa/mesa_git.bb:28]: Ok, done.

**Pull request merged**